docker_daemon/check.py

```diff
@@ -472,17 +472,28 @@ def _extract_tag_value(self, entity, tag_name):
 
         return entity["_tag_values"][tag_name]
 
+    def _localhost_ecs_agent(self):
+        result = ""
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(5)
+        result = sock.connect_ex(('localhost', ECS_INTROSPECT_DEFAULT_PORT))
```

Comment:
this should be in a `try... except` block as it can raise exception. And we could log that exception on debug level, as well as `result` if it's `!= 0` (result contains C errors, exceptions are Python level failures).